src/utils/wallet.ts

```diff
@@ -0,0 +1,220 @@
+      const wordsArray = newWordsString.split(' ');
+      if (wordsArray.length !== 24) {
+        // Must have 24 words
+        return {'valid': false, 'message': 'Must have 24 words'};
+      } else if (!Mnemonic.isValid(newWordsString)) {
+        // Check if there is a word that does not belong to the list of possible words
+        const wordlist = Mnemonic.Words.ENGLISH;
+        const errorList: string[] = [];
+
+        for (const word of wordsArray) {
+          if (wordlist.indexOf(word) < 0) {
+            errorList.push(word);
+          }
+        }
+
+        let errorMessage = '';
+        if (errorList.length > 0) {
+          errorMessage = `Invalid words: ${errorList.join(' ')}`;
+        } else {
+          // Invalid sequence of words
+          errorMessage = 'Invalid sequence of words';
+        }
+        return {'valid': false, 'message': errorMessage};
+      }
+    } else {
+      // Must be string
+      return {'valid': false, 'message': 'Must be a string'};
+    }
+    return {'valid': true, 'message': '', 'words': newWordsString};
+  },
+
+  /**
+   * Generate HD wallet words
+   *
+   * @param {number} entropy Data to generate the HD Wallet seed - entropy (256 - to generate 24 words)
+   *
+   * @return {string} words generated
+   * @memberof Wallet
+   * @inner
+   */
+  generateWalletWords(entropy: number = HD_WALLET_ENTROPY): string {
+    const code = new Mnemonic(entropy);
+    return code.phrase;
+  },
+
+  /**
+   * Get xpub from data
+   *
+   * @param {Buffer} pubkey Compressed public key
+   * @param {Buffer} chainCode HDPublic key chaincode
+   * @param {Buffer} fingerprint parent fingerprint
+   * @param {string} networkName Optional parameter to select the used network (default is mainnet)
+   *
+   * @return {String} Xpub
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  xpubFromData(pubkey: Buffer, chainCode: Buffer, fingerprint: Buffer, networkName: string = 'mainnet'): string {
+    const network = new Network(networkName);
+    const hdpubkey = new HDPublicKey({
+      network: network.bitcoreNetwork,
+      depth: 4,
+      parentFingerPrint: fingerprint,
+      childIndex: 0,
+      chainCode: chainCode,
+      publicKey: pubkey
+    });
+
+    return hdpubkey.xpubkey;
+  },
+
+  /**
+   * Get compressed public key from uncompressed
+   *
+   * @param {Buffer} pubkey Uncompressed public key
+   *
+   * @return {Buffer} Compressed public key
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  toPubkeyCompressed(pubkey: Buffer): Buffer {
+    const x = pubkey.slice(1, 33);
+    const y = pubkey.slice(33, 65);
+    const point = new crypto.Point(x, y);
+    return crypto.Point.pointToCompressed(point);
+  },
+
+  /**
+   * Get public key for specific key index derivation
+   *
+   * @param {number} index Index of the key to derive
+   *
+   * @return {Buffer} Public key
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  getPublicKeyFromXpub(xpubkey: string, index: number): Buffer {
```

Comment:
Should we receive a path?

Reply:
This method is a bit different. Here we expect the `xpub` of the specific key you want to derive. I think a method that we receive a root xpub and derive the path is valid but should be discussed. Should we accept a different an xpub from a different source than the root? Should we accept a partial path?

I think we will need a method like this eventually and then we can add. What do you think?

Reply:
There's no way to know whether the xpub is the root or a derived path. I agree with the current format. Maybe we should just update the docs to let the user know that they should derivate the path and send the xpub just after doing it.

Reply:
Done